File: frontend/types/uxlab.ts
export type FindingSeverity = 'critical' | 'major' | 'minor';
export type FindingType    = 'ISSUE' | 'WARNING' | 'POSITIVE';
export type FindingStatus  = 'open' | 'patched' | 'dismissed';

export interface CompetitorEvidence {
  url: string;
  screenshotUrl: string;
  annotation: string; // one sentence: what the competitor does differently
}

export interface Finding {
  id: string;
  type: FindingType;
  severity: FindingSeverity;
  status: FindingStatus;
  component: string;              // dot-separated path e.g. "hero.cta"
  title: string;                  // max 8 words
  description: string;
  principle: string;
  principleExplanation: string;
  recommendation: string;
  requiresCompetitorEvidence: boolean;
  competitorEvidence?: CompetitorEvidence[];
  annotation: { xPercent: number; yPercent: number };
}

export interface UXLabSession {
  id: string;
  url: string;
  page: string;
  beforeScreenshotUrl: string;
  afterScreenshotUrl: string;
  afterPreviewMessage?: string;
  findings: Finding[];
  createdAt: string;
  status: 'pending' | 'complete' | 'error';
}

File: backend/app/schemas/ux_lab.py
from __future__ import annotations

from typing import Optional
from pydantic import BaseModel


class AnalyzeRequest(BaseModel):
    repo_url: str
    branch: str = "main"
    page: str = "/"
    access_token: str = ""
    workspace_id: str = "anonymous"
    analysis_only: bool = False


class FindingOut(BaseModel):
    id: str
    type: str                   # ISSUE | WARNING | POSITIVE
    severity: str               # critical | major | minor
    status: str                 # open | patched | dismissed
    component: str
    title: str
    description: str
    principle: str
    principle_explanation: str
    recommendation: str
    annotation: dict            # {xPercent, yPercent}


class SessionOut(BaseModel):
    id: str
    url: str
    page: str
    before_screenshot_url: str
    after_screenshot_url: str
    after_preview_message: Optional[str] = None
    findings: list[FindingOut]
    created_at: str
    status: str

    class Config:
        from_attributes = True


class ApplyFindingRequest(BaseModel):
    finding_id: str
